conductor/blueprints/search/models.py

```diff
@@ -0,0 +1,57 @@
+import os
+import json
+
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import NotFoundError
+
+_engine = None
+
+ENABLED_SEARCHES = {
+    'user': {
+        'index': 'users',
+        'doc_type': 'user_profile',
+        '_source': ['idhash', 'name', 'avatar_url', 'datasets']
+    },
+    'package': {
+        'index': 'packages',
+        'doc_type': 'package',
+        '_source': ['id', 'model', 'package', 'origin_url']
+    }
+}
+
+def _get_engine():
+    global _engine
+    if _engine is None:
+        es_host = os.environ['OS_ELASTICSEARCH_ADDRESS']
+        _engine = Elasticsearch(hosts=[es_host], use_ssl='https' in es_host)
+    return _engine
+
+
+def query(kind, size=100, **kw):
+    kind_params = ENABLED_SEARCHES.get(kind)
+    if kind_params is None:
+        return None
+    try:
+        if type(size) is list:
+            size = size[0]
+        api_params = {'size': int(size)}
+        api_params.update(kind_params)
+        q_str = ") AND (".join([" OR ".join(["{0}:{1}".format(k, json.dumps(json.loads(v)))
+                                             for v in v_arr])
+                                for k, v_arr in kw.items()
+                                if k.split('.')[0] in kind_params['_source']])
+        if len(q_str) == 0:
+            q_str = "{0}:*".format(kind_params['_source'][0])
```

Comment:
I'm not sure if it'll work in your case, but an empty string (or even `undefined`) returns everything when using the elasticsearch JS library. If it's the same for Python, then it'll allow you to remove (or simplify) this line.